lib/collection/src/common/stoppable_task.rs

```diff
@@ -1,12 +1,16 @@
+use std::any::Any;
 use std::sync::atomic::{AtomicBool, Ordering};
 use std::sync::{Arc, Weak};
 
 use tokio::task::JoinHandle;
 
+type PanicPayload = Box<dyn Any + Send + 'static>;
+
 pub struct StoppableTaskHandle<T> {
     pub join_handle: JoinHandle<Option<T>>,
     started: Arc<AtomicBool>,
     stopped: Weak<AtomicBool>,
+    panic_handler: Option<Box<dyn Fn(PanicPayload) + Sync + Send>>,
```

Comment:
Why not making the `panic_handler` required?
Would that make the PR too large due to all the call sites?

Reply:
Could do. I'm not afraid of it becoming too large. But most of the calls will just have an empty handler.

Also, regular threads don't require (or support) a panic handler either. Most of the time, these will just panic silently.

Reply:
Let's keep it like that :+1: